File: app/src/main/java/fr/piotr/dismoitoutsms/util/Instruction.java
package fr.piotr.dismoitoutsms.util;

/**
 * Created by piotr_000 on 20/03/2016.
 *
 */
public enum Instruction {

    REPONDRE_FERMER, REPONSE, MODIFIER_ENVOYER_FERMER, DICTER_CONTACT;

    public boolean is(Instruction ... instructions){
        for (Instruction instruction : instructions) {
            if(this==instruction){
                return true;
            }
        }
        return false;
    }
}

File: app/src/main/java/fr/piotr/dismoitoutsms/util/Instruction.kt
package fr.piotr.dismoitoutsms.util

/**
 * Created by piotr_000 on 20/03/2016.
 *
 */
enum class Instruction {

    LIRE_FERMER, REPONDRE_FERMER, REPONSE, AJOUTER, MODIFIER_ENVOYER_FERMER, DICTER_CONTACT;

    fun `is`(vararg instructions: Instruction): Boolean {
        return instructions.any { this == it }
    }
}
